client/src/accounts/authentication/register.hook.test.tsx

```diff
@@ -0,0 +1,117 @@
+
+interface Result {
+  id: Number;
+  error?: { message: string };
+  mutate: (form: Record<string, unknown>) => void;
+  isError: boolean;
+  isSuccess: boolean;
+  data?: Record<string, unknown>;
+}
+
+describe('useRegisterHook', () => {
+  it('should throw an error if any fields are missing', async () => {
+    // we're missing the confirm password
+    const expectedMessage = 'Please re-authenticate';
+    server.use(
+      rest.post(ENDPOINT, (req, res, ctx) => res(ctx.status(HTTP_BAD_REQUEST), ctx.json({ message: expectedMessage }))),
+    );
+
+    const registerForm: RegistrationFormType = {
+      email: 'bob@example.com',
+      firstName: 'Bob',
+      lastName: 'Ross',
+      password: 'otherpassword',
+      confirmPassword: 'otherpassword',
+    };
+    const { result } = renderHook<Result, RegistrationFormType>(() => useRegister());
+
+    act(() => {
+      result.current.mutate(registerForm);
+    });
+    await waitFor(() => {
+      expect(result.current.isError).toBe(true);
+      expect(result.current.error).toEqual({ message: expectedMessage });
+    });
+  });
+
+  it('should throw an error if email is malformed', async () => {
+    // not a well-formed email address
+    // we're missing the confirm password
+    const expectedMessage = 'Please re-authenticate';
+    server.use(
+      rest.post(ENDPOINT, (req, res, ctx) =>
+        res(ctx.status(HTTP_BAD_REQUEST), ctx.json({ error: { message: expectedMessage } })),
+      ),
+    );
+
+    const registerForm: RegistrationFormType = {
+      email: 'bob@example',
+      firstName: 'Bob',
+      lastName: 'Ross',
+      password: 'otherpassword',
+      confirmPassword: 'otherpassword',
+    };
+    const { result } = renderHook<Result, RegistrationFormType>(() => useRegister());
+
+    act(() => {
+      result.current.mutate(registerForm);
+    });
+    await waitFor(() => {
+      expect(result.current.isError).toBe(true);
+      expect(result.current.error).toEqual({ message: expectedMessage });
+    });
+  });
+
+  it('should throw an error if email has already been registered', async () => {
+    // attempt to reuse existing email address
+    const expectedMessage = 'Please re-authenticate';
+    server.use(
+      rest.post(ENDPOINT, (req, res, ctx) => res(ctx.status(HTTP_BAD_REQUEST), ctx.json({ message: expectedMessage }))),
+    );
+
+    const registerForm: RegistrationFormType = {
+      email: 'bob@example.com',
+      firstName: 'Bob',
+      lastName: 'Ross',
+      password: 'notreallybob',
+      confirmPassword: 'notreallybob',
+    };
+    const { result } = renderHook<Result, RegistrationFormType>(() => useRegister());
+
+    act(() => {
+      result.current.mutate(registerForm);
+    });
+    await waitFor(() => expect(result.current.isError).toBe(true));
+    expect(result.current.error).toEqual({ message: expectedMessage });
+  });
+
+  it('should return if registration was successful', async () => {
+    // happy path
```

Comment:
ths comment helps no one, so remove. What should it return if successful, it does return something doesn't it? If not then  say something like `should return successful response if registration completed`.